kolibri/plugins/coach/assets/src/state/actions/exam.js

```diff
@@ -416,22 +419,69 @@ function createExam(store, classCollection, examObj) {
   );
 }
 
-function showExamReportPage(store, classId, examId) {
+function showExamReportPage(store, classId, channelId, examId) {
   store.dispatch('CORE_SET_PAGE_LOADING', true);
   store.dispatch('SET_PAGE_NAME', Constants.PageNames.EXAM_REPORT);
-  store.dispatch('SET_PAGE_STATE', { classId });
-  store.dispatch('CORE_SET_ERROR', null);
-  store.dispatch('CORE_SET_TITLE', ('Exam Report'));
-  store.dispatch('CORE_SET_PAGE_LOADING', false);
+  const examLogPromise = ExamLogResource.getCollection({
+    exam: examId,
+    collection: classId,
+  }).fetch();
+  const examPromise = ExamResource.getModel(examId, { channel_id: channelId }).fetch();
+  const facilityUserPromise = FacilityUserResource.getCollection({ member_of: classId }).fetch();
+  const groupPromise = LearnerGroupResource.getCollection({ parent: classId }).fetch();
+  ConditionalPromise.all([examLogPromise, facilityUserPromise, groupPromise, examPromise]).only(
+    CoreActions.samePageCheckGenerator(store),
+    ([examLogs, facilityUsers, learnerGroups, exam]) => {
+      console.log(facilityUsers, examLogs);
+      const examTakers = facilityUsers.map(
+      (user) => {
+        const examTakenByUser = examLogs.find((examLog) => String(examLog.user) === user.id) || {};
+        const learnerGroup = learnerGroups.find(
+          group => group.user_ids.indexOf(user.id) > -1) || {};
+        return {
+          id: user.id,
+          name: user.full_name,
+          group: learnerGroup,
+          score: examTakenByUser.score,
+          progress: examTakenByUser.progress,
+        };
+      });
+      const pageState = {
+        examTakers,
+        classId,
+        exam,
+        channelId,
+      };
+      store.dispatch('SET_PAGE_STATE', pageState);
+      store.dispatch('CORE_SET_ERROR', null);
+      store.dispatch('CORE_SET_TITLE', ('Exam Report'));
+      store.dispatch('CORE_SET_PAGE_LOADING', false);
+    },
+    error => { CoreActions.handleApiError(store, error); }
```

Comment:
I would argue that because the intention of this one liner is to execute a function, not return a value, then the use of curly braces is more semantic - but yes, not strictly necessary syntactically.